go/vt/vtgate/evalengine/translate.go

```diff
@@ -402,20 +402,32 @@ func translateUnaryExpr(unary *sqlparser.UnaryExpr, lookup TranslationLookup) (E
 	}
 }
 
-func translateConvertCharset(charset string, lookup TranslationLookup) (collations.ID, error) {
+func binaryCollationForCollation(collation collations.ID) collations.ID {
+	binary := collations.Local().LookupByID(collation)
+	return collations.Local().BinaryCollationForCharset(binary.Charset().Name()).ID()
```

Comment:
I feel like we may want to ensure that:
```go
collations.Local().BinaryCollationForCharset(binary.Charset().Name())
```
is not returning `nil` before calling `ID()`.

Reply:
if the given collation ID does not have a binary equivalent it would return `nil`

Reply:
@frouioui Yeah, can add some safety checks. I think in practice though, every charset and thus also every charset through a collation has a binary collation available. It's always `<charset>_bin`. But a check won't hurt here and good to have just in case.

Reply:
Done!